### DISCUSSION TITLE
Question: running multiple "process" beans on batch mode

### DISCUSSION THREAD

**Discussion by fmertins** [forcedotcom/dataloader#738] [Q&A] [answered]
Hello, I'm trying to run DL on the batch mode in a way that it would just consider all the `<bean>` elements that are configured on the `process-conf.xml` file, is that possible?

I've been reading the documentation and samples but so far no luck. For instance, I've created two beans like:

```
<beans>
    <bean id="account-extract" class="com.salesforce.dataloader.process.ProcessRunner" scope="prototype">
    ...
    </bean>

    <bean id="account-insert" class="com.salesforce.dataloader.process.ProcessRunner" scope="prototype">
    ...
    </bean>
</beans>
```

Then, upon running DL I got this:

**C:\Users\Fernando_Mertins\Some_path>"C:\Data Loader\bin\process.bat" .**

_2022-11-07 11:41:09,271 INFO  [main] process.ProcessRunner getInstance (ProcessRunner.java:324) - process.nameis not specified in the command line. Loading the process properties from config.properties._

Since I didn't inform that second `<batch process bean id>` optional parameter, I was expecting that DL would just consider the entire XML file.

Thank you in advance!

**Comment by ashitsalesforce** [comment, marked as answer]
Hi @fmertins ,

bean id is an optional parameter because it can be specified as a command line option or can be specified in config.properties file according to [the documentation](https://developer.salesforce.com/docs/atlas.en-us.dataLoader.meta/dataLoader/using_the_command_line_interface.htm):
<batch process bean id> (optional) The id of the batch process bean of class com.salesforce.dataloader.process.ProcessRunner defined in the process-conf.xmlfile. If not provided, then the value of the process.name property in the config.properties file is used.

Data loader executes a single operation in batch mode. The operation is specified by its process id. If you want to execute multiple operations specified in your process-conf.xml, you can write a wrapper script to invoke process.bat multiple times, each time with a different process id.

**Comment by fmertins** [comment]
Hey there!

> Data loader executes a single operation in batch mode. The operation is specified by its process id. If you want to execute multiple operations specified in your process-conf.xml, you can write a wrapper script to invoke process.bat multiple times, each time with a different process id.

Got it, thank you! I was assuming the "batch" concept here would be the capability of having multiple beans, and then the tool would process them, in a batch... :-)
